staging/src/github.com/clusterpedia-io/api/cluster/v1alpha2/types.go

```diff
@@ -107,6 +107,10 @@ type ClusterGroupResources struct {
 }
 
 type ClusterStatus struct {
+	// +required
+	// +kubebuilder:validation:Required
+	APIServer string `json:"apiserver,omitempty"`
```

Comment:
The required field cannot have an omitmepty tag, otherwise it is not a required field in the generated CRD.
Also would it be better to allow apiserver to be empty, as an optional field?

The following Version should be an optional field, submit a new PR to fix it if you wish.

Reply:
updated 
or we can make a good-first-issue with `Version`

Reply:
good idea